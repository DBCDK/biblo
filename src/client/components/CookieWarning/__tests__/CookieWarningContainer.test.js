/**
 * @file Testing the CookieWarningContainer component
 */

import React from 'react';
import {shallow} from 'enzyme';
import Cookies from 'js-cookie';

import CookieWarningContainer from '../CookieWarningContainer.component';

describe('Testing the CookieWarningContainer component', () => {

  it('Should render a modal', () => {
    Cookies.remove('reddi-fe-cookie');
    const tree = shallow(<CookieWarningContainer />);

    expect(tree.find('.cookie-warning')).toHaveLength(1); // eslint-disable-line no-undef
    expect(tree).toMatchSnapshot(); // eslint-disable-line no-undef
  });

  it('Should hide the cookie warning', () => {
    const tree = shallow(<CookieWarningContainer />);
    const instance = tree.instance();

    expect(instance.state.displayWarning).toBe(true); // eslint-disable-line no-undef
    instance.onClose();
    tree.update();

    expect(instance.state.displayWarning).toBe(false); // eslint-disable-line no-undef
    expect(tree.find('.cookie-warning')).toHaveLength(0); // eslint-disable-line no-undef
  });
});
