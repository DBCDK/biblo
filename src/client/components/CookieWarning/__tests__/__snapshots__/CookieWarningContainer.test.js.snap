// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CookieWarningContainer component Should render a modal 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CookieWarningContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="cookie-warning"
      >
        <div
          className="cookie-warning--svg"
        >
          <Icon
            className="icon"
            glyph="test-file-stub"
            height={30}
            width={100}
          />
        </div>
        <span
          className="cookie-warning--header"
        >
          Velkommen til biblo.dk
        </span>
        <span
          className="cookie-warning--message"
        >
          For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.
        </span>
        <RoundedButton.a.component
          buttonText="OK"
          className=""
          clickFunction={[Function]}
          compact={false}
          href="#!"
        />
      </div>,
      "onClose": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cookie-warning--svg"
          >
            <Icon
              className="icon"
              glyph="test-file-stub"
              height={30}
              width={100}
            />
          </div>,
          <span
            className="cookie-warning--header"
          >
            Velkommen til biblo.dk
          </span>,
          <span
            className="cookie-warning--message"
          >
            For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.
          </span>,
          <RoundedButton.a.component
            buttonText="OK"
            className=""
            clickFunction={[Function]}
            compact={false}
            href="#!"
          />,
        ],
        "className": "cookie-warning",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              className="icon"
              glyph="test-file-stub"
              height={30}
              width={100}
            />,
            "className": "cookie-warning--svg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "icon",
              "glyph": "test-file-stub",
              "height": 30,
              "width": 100,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Velkommen til biblo.dk",
            "className": "cookie-warning--header",
          },
          "ref": null,
          "rendered": "Velkommen til biblo.dk",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.",
            "className": "cookie-warning--message",
          },
          "ref": null,
          "rendered": "For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "OK",
            "className": "",
            "clickFunction": [Function],
            "compact": false,
            "href": "#!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="cookie-warning"
        >
          <div
            className="cookie-warning--svg"
          >
            <Icon
              className="icon"
              glyph="test-file-stub"
              height={30}
              width={100}
            />
          </div>
          <span
            className="cookie-warning--header"
          >
            Velkommen til biblo.dk
          </span>
          <span
            className="cookie-warning--message"
          >
            For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.
          </span>
          <RoundedButton.a.component
            buttonText="OK"
            className=""
            clickFunction={[Function]}
            compact={false}
            href="#!"
          />
        </div>,
        "onClose": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cookie-warning--svg"
            >
              <Icon
                className="icon"
                glyph="test-file-stub"
                height={30}
                width={100}
              />
            </div>,
            <span
              className="cookie-warning--header"
            >
              Velkommen til biblo.dk
            </span>,
            <span
              className="cookie-warning--message"
            >
              For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.
            </span>,
            <RoundedButton.a.component
              buttonText="OK"
              className=""
              clickFunction={[Function]}
              compact={false}
              href="#!"
            />,
          ],
          "className": "cookie-warning",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className="icon"
                glyph="test-file-stub"
                height={30}
                width={100}
              />,
              "className": "cookie-warning--svg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "icon",
                "glyph": "test-file-stub",
                "height": 30,
                "width": 100,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Velkommen til biblo.dk",
              "className": "cookie-warning--header",
            },
            "ref": null,
            "rendered": "Velkommen til biblo.dk",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.",
              "className": "cookie-warning--message",
            },
            "ref": null,
            "rendered": "For at gøre siden så let at bruge som muligt gemmer vi det du laver på sitet i noget, der hedder en cookie.",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonText": "OK",
              "className": "",
              "clickFunction": [Function],
              "compact": false,
              "href": "#!",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
